### ISSUE TITLE
Weblate Docker image not working

### ISSUE THREAD

**Issue by rwrx** [WeblateOrg/weblate#2615]
I have pulled newest Docker image of Weblate on my VPS and I am getting this error:

```
weblate_1   | During handling of the above exception, another exception occurred:
weblate_1   |
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/local/bin/weblate", line 11, in <module>
weblate_1   |     sys.exit(main())
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/weblate/runner.py", line 34, in main
weblate_1   |     execute_from_command_line(argv)
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
weblate_1   |     utility.execute()
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 375, in execute
weblate_1   |     self.fetch_command(subcommand).run_from_argv(self.argv)
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 316, in run_from_argv
weblate_1   |     self.execute(*args, **cmd_options)
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 353, in execute
weblate_1   |     output = self.handle(*args, **options)
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/weblate/utils/management/commands/cleanup_celery.py", line 43, in handle
weblate_1   |     scheduler._remove_db()
weblate_1   | UnboundLocalError: local variable 'scheduler' referenced before assignment
weblate_1   | Exception ignored in: <bound method Shelf.__del__ of <shelve.DbfilenameShelf object at 0x7f44d9a653c8>>
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 162, in __del__
weblate_1   |     self.close()
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 144, in close
weblate_1   |     self.sync()
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 168, in sync
weblate_1   |     self[key] = entry
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 125, in __setitem__
weblate_1   |     self.dict[key.encode(self.keyencoding)] = f.getvalue()
weblate_1   | _dbm.error: cannot add item to database
weblate_1   | Postgres is up
weblate_1   | Operations to perform:
weblate_1   |   Apply all migrations: accounts, addons, admin, auth, authtoken, checks, contenttypes, gitexport, lang, memory, screenshots, sessions, sites, social_django, trans, weblate_auth, wladmin
weblate_1   | Running migrations:
weblate_1   |   No migrations to apply.
weblate_1   | Removing corrupted schedule file: error('cannot add item to database',)
weblate_1   | ERROR Handled exception UnboundLocalError: local variable 'scheduler' referenced before assignment
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 111, in __getitem__
weblate_1   |     value = self.cache[key]
weblate_1   | KeyError: 'entries'
weblate_1   |
weblate_1   | During handling of the above exception, another exception occurred:
weblate_1   |
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 515, in _create_schedule
weblate_1   |     self._store[str('entries')]
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 113, in __getitem__
weblate_1   |     f = BytesIO(self.dict[key.encode(self.keyencoding)])
weblate_1   | KeyError: b'entries'
weblate_1   |
weblate_1   | During handling of the above exception, another exception occurred:
weblate_1   |
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/weblate/utils/management/commands/cleanup_celery.py", line 36, in handle
weblate_1   |     app=app
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 458, in __init__
weblate_1   |     Scheduler.__init__(self, *args, **kwargs)
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 225, in __init__
weblate_1   |     self.setup_schedule()
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 486, in setup_schedule
weblate_1   |     self._create_schedule()
weblate_1   |   File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 519, in _create_schedule
weblate_1   |     self._store[str('entries')] = {}
weblate_1   |   File "/usr/lib/python3.6/shelve.py", line 125, in __setitem__
weblate_1   |     self.dict[key.encode(self.keyencoding)] = f.getvalue()
weblate_1   | _dbm.error: cannot add item to database
```

How can I resolve it? Thank you a lot.

**Comment by nijel** [comment]
I thought this has been finally addressed, see https://github.com/WeblateOrg/docker/issues/105 ....

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).

**Comment by rwrx** [comment]
Thank you. But I am still getting this error. I have latest weblate:3.5-1 Docker image. I have tried to run:

`sudo docker-compose run --rm weblate cleanup_celery`

and I got this output:

```
Starting weblatedocker_cache_1 ...
Starting weblatedocker_database_1 ...
Starting weblatedocker_cache_1
Starting weblatedocker_database_1 ... done
Postgres is up
Removing corrupted schedule file: error('cannot add item to database',)
ERROR Handled exception UnboundLocalError: local variable 'scheduler' referenced before assignment
Traceback (most recent call last):
  File "/usr/lib/python3.6/shelve.py", line 111, in __getitem__
    value = self.cache[key]
KeyError: 'entries'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 515, in _create_schedule
    self._store[str('entries')]
  File "/usr/lib/python3.6/shelve.py", line 113, in __getitem__
    f = BytesIO(self.dict[key.encode(self.keyencoding)])
KeyError: b'entries'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/weblate/utils/management/commands/cleanup_celery.py", line 36, in handle
    app=app
  File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 458, in __init__
    Scheduler.__init__(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 225, in __init__
    self.setup_schedule()
  File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 486, in setup_schedule
    self._create_schedule()
  File "/usr/local/lib/python3.6/dist-packages/celery/beat.py", line 519, in _create_schedule
    self._store[str('entries')] = {}
  File "/usr/lib/python3.6/shelve.py", line 125, in __setitem__
    self.dict[key.encode(self.keyencoding)] = f.getvalue()
_dbm.error: cannot add item to database

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/weblate", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/weblate/runner.py", line 34, in main
    execute_from_command_line(argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/dist-packages/weblate/utils/management/commands/cleanup_celery.py", line 43, in handle
    scheduler._remove_db()
UnboundLocalError: local variable 'scheduler' referenced before assignment
Exception ignored in: <bound method Shelf.__del__ of <shelve.DbfilenameShelf object at 0x7f54f7282978>>
Traceback (most recent call last):
  File "/usr/lib/python3.6/shelve.py", line 162, in __del__
    self.close()
  File "/usr/lib/python3.6/shelve.py", line 144, in close
    self.sync()
  File "/usr/lib/python3.6/shelve.py", line 168, in sync
    self[key] = entry
  File "/usr/lib/python3.6/shelve.py", line 125, in __setitem__
    self.dict[key.encode(self.keyencoding)] = f.getvalue()
_dbm.error: cannot add item to database
```

I am not that familiar with Weblate so currently I don't know how to fix this. Thank you.

**Comment by nijel** [comment]
I've fixed this for upcoming 3.5.1 release, it will get to the docker image as well.

**Comment by nijel** [comment]
The fixed 3.5.1-1 image is now out.

**Comment by rwrx** [comment]
Thank you a lot, I can confirm that it is fixed now.
